packages/runtime/src/widgets/Search.tsx

```diff
@@ -63,55 +56,48 @@ export const Search: React.FC<SearchProps> = ({
     name: itemTemplate?.name,
   });
 
-  const { rootRef, values } = useRegisterBindings(
-    { styles, value, options, ...rest, widgetName },
-    id,
+  const evaluatedNamedData = useEvaluate({ namedData });
+
+  const { id, rootRef, values } = useRegisterBindings(
+    { styles, value, ...rest, widgetName },
+    rest.id,
     {
       setValue,
-      setOptions,
     },
   );
 
   const onSearchAction = useEnsembleAction(onSearch);
   const onChangeAction = useEnsembleAction(onChange);
   const onSelectAction = useEnsembleAction(onSelect);
 
-  useEffect(() => {
-    const newOptions = namedData.map((item) => {
-      const itemData = itemTemplate?.name
-        ? (get(item, itemTemplate.name) as unknown)
-        : item;
-
-      const optionValue = get(itemData, searchKey ?? "") as unknown;
-      const option = {
-        data: itemData,
-        value: optionValue,
-        label: itemTemplate?.template ? (
-          <SearchOption
-            context={item as CustomScope}
-            template={itemTemplate.template}
-          />
-        ) : (
-          JSON.stringify(optionValue)
-        ),
-      };
-      return option;
-    });
-    setOptions(newOptions);
-  }, [itemTemplate, namedData, searchKey, setOptions, value]);
-
-  const filteredOptions = useMemo(() => {
-    // We assume the search callback does the result filtering
-    if (onSearchAction?.callback) {
-      return options;
-    }
-    if (!searchValue?.trim().length) {
-      return options;
+  // rendered options
+  const renderOptions = useMemo(() => {
+    let dropdownOptions = null;
+
+    if (isObject(itemTemplate) && !isEmpty(evaluatedNamedData.namedData)) {
+      const tempOptions = evaluatedNamedData.namedData.map((item: unknown) => {
+        const optionValue = get(item, [itemTemplate.name, searchKey || ""]) as
+          | string
+          | number;
```

Comment:
Should use item.value here if searchKey doesn't exit?

Reply:
@evshi Instead of using item.value we need to use itemTemplate.value, this will helpful us when there is only array of string.